### PR TITLE
 Reset user  password  when forgoten

### PR DESCRIPTION
**Pull request by Egide138**
--
### DESCRIPTION OF TASK TO BE COMPLETED
- User can request for password reset if he/she forgot the one he/she was using
- An email containing  a reset link will be sent to that user 
- By using that link he can update his/her password
---
### HOW SHOULD THIS BE MANUALLY TESTED?
**Testing** `/api/route` 
1. clone this REPO
1. Go to (checkout ) this branch `ft-reset-password-TlqFewPw`
1. Do `npm install` to install 
1. Make a `.env` file and fill it with key and values  as referenced  in the `.ev_example` file.
1. Run `npm run dev` to run in development mode
---
#### Testing using postman

- Make a POST request to route `/api/users/forgot` along with data **email** and password of user you want to reset his/her password
- 
- Use the link you get in response or in mailbox and make a `PATCH` request on it with **email** and **new password**.
--- 
### RELEVANT TRELLO CARD
ID : TlqFewPw

### CODE REVIEW

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -120,5 +120,45 @@ export const signin = async (req, res) => {
...
+  
+  const user=await User.findOne({where:{email:email}})
+  if(!user){
+    return errorRes(res,500,res.__('Sorry!No user found with that email address.'))
+  }else{
+    const newPassword  = await bcrypt.hash(password,10, async(err,result)=>{
+      if(result){
+     await  User.update({password:newPassword},{where:{id:user.id}})
+
+    //  await sendEmail('resetPassword',{email:user.email})
```
Comment (samuel776) [inline src/controllers/usersController.js]: if this commented line is no longer in use should be removed

**Review by samuel776 — COMMENTED** [review]
great, i suggest that you can squash these commits into one and resolve the conflicts

**Comment by ceelogre** [comment]
Your PR has no description and the title needs to be more descriptive.

File: src/controllers/usersController.js (hunk 1 above)
Comment (Egide138) [reply src/controllers/usersController.js]: thank you

File: .eslintrc.json [hunk 2]
```diff
@@ -9,7 +9,11 @@
     "sourceType": "module"
   },
   "rules": {
+<<<<<<< HEAD
     "no-console": "off",
+=======
+    "no-console": "off"
+>>>>>>> 91afe82 (feat(authentication):signin)
```
Comment (KundwaBruno) [inline .eslintrc.json]: Great work Egide, Try to resolve these conflicts

**Review by 41hulk — COMMENTED** [review]
Good Job Egide but the conflicts must be solved

File: .eslintrc.json (hunk 2 above)
Comment (Egide138) [reply .eslintrc.json]: Thanks

**Review by mansurissa — APPROVED** [review]
Good job Egide

File: src/controllers/locales/en.json [hunk 3]
```diff
@@ -1,36 +1,21 @@
...
+	"Successfully got All roles": "Successfully got All roles",
+	"Welcome to phantom an app which is used to track buses": "Welcome to phantom an app which is used to track buses",
+	"Successfully got All users": "Successfully got All users",
+	"Successfully got All drivers": "Successfully got All drivers",
+	"User created Successfully and email was sent": "User created Successfully and email was sent",
+	"Validation error: lastName is required": "Validation error: lastName is required",
+	"Successfully verfied your Email.": "Successfully verfied your Email.",
+	"Logged out successfully": "Logged out successfully",
+	"Validation error: email must be a valid email": "Validation error: email must be a valid email"
+}
```
Comment (raymond42) [inline src/controllers/locales/en.json]: add an empty line here

Comment (mansurissa) [reply src/controllers/locales/en.json]: yeah

**Review by mansurissa — APPROVED** [review]

**Comment by 41hulk** [comment]
LGTM

**Review by KundwaBruno — APPROVED** [review]

**Review by raymond42 — APPROVED** [review]

**Pull request merged**